marshmallow/marshalling.py

```diff
@@ -275,6 +283,22 @@ def deserialize(self, data, fields_dict, many=False, partial=False,
             if value is not missing:
                 key = fields_dict[attr_name].attribute or attr_name
                 set_value(ret, key, value)
+
+        if unknown != IGNORE:
+            for key in set(data) - set(fields_dict):
+                value = data[key]
+                if unknown == ALLOW:
+                    set_value(ret, key, value)
+                elif unknown == RAISE:
+                    def _raise(_):
+                        raise ValidationError('Unkown key.')
+                    self.call_and_store(
+                        getter_func=_raise,
+                        data=value,
+                        field_name=key,
+                        field_obj=type(value)
+                    )
```

Comment:
I'm not very happy that I used `call_and_store` in a hackish way, but I could not find a better solution to achieve the same thing.

Reply:
Hm, maybe we should factor out the error storage code into its own method, so this could look something like:

```python
self.store_error(
    'Unknown field.',
    data=value,
    field_name=key,
)
```

Reply:
Thank you guys, I was a bit to shy to change this but that's indeed a better way of doing it.

Reply:
Done. What do you both think of it that way?